benchmarks/benchmarks/utils.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+import gc
+import sys
+from functools import wraps
+from string import ascii_lowercase
+from time import sleep
+from typing import TYPE_CHECKING, Literal, TypeVar, get_args
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from memory_profiler import memory_usage
+from scipy import sparse
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Set
+
+    from numpy.typing import NDArray
+
+    OneDIndex = NDArray[np.bool_] | NDArray[np.intp] | slice
+    C = TypeVar("C", bound=Callable)
+
+
+def get_actualsize(input_obj: object) -> int:
+    """Using Python Garbage Collector to calculate the size of all elements attached to an object"""
+
+    memory_size = 0
+    ids = set()
+    objects = [input_obj]
+    while objects:
+        new = []
+        for obj in objects:
+            if id(obj) not in ids:
+                ids.add(id(obj))
+                memory_size += sys.getsizeof(obj)
+                new.append(obj)
+        objects = gc.get_referents(*new)
+    return memory_size
+
+
+def get_anndata_memsize(adata: AnnData) -> float:
```

Comment:
Is this function used anywhere?

Reply:
no, I’ll remove it. Also a candidate to move to anndata’s suite